### PR TITLE
Upgrade to ts http runtime

### PR DESCRIPTION
**Pull request by MaryGao**
fixes https://github.com/Azure/autorest.typescript/issues/2081

- Enable the emitter option `branded` and allow to generate the non-branded SDK
    - Enable smoke test for OpenAI
    - Enable integration testing for model useage for non-branded
- Other small fixes 
   - not generate credential info when addCredential is false
   - fix the description is undefined in package.json

### CODE REVIEW

File: packages/typespec-test/test/openai_non_branded/generated/typespec-ts/README.md
Comment (MaryGao) [inline packages/typespec-test/test/openai_non_branded/generated/typespec-ts/README.md:20]: Metadata files improvement should be covered in issue https://github.com/Azure/autorest.typescript/issues/2100

File: packages/typespec-ts/package.json
Comment (MaryGao) [inline packages/typespec-ts/package.json:74]: @joheredi Do we have a plan to review a preview version for runtime lib before Nov Demo?

File: packages/typespec-test/test/openai_non_branded/generated/typespec-ts/package.json
Comment (MaryGao) [inline packages/typespec-test/test/openai_non_branded/generated/typespec-ts/package.json:2]: There will be no `azure` wordings could be searched in this folder.

File: packages/typespec-test/test/parametrizedHost/generated/typespec-ts/src/parametrizedHostClient.ts
Comment (MaryGao) [inline packages/typespec-test/test/parametrizedHost/generated/typespec-ts/src/parametrizedHostClient.ts:22]: Change is introduced by addCredential is false fix.

File: packages/typespec-ts/src/index.ts
Comment (joheredi) [inline packages/typespec-ts/src/index.ts:225]: I was hoping we could calculate this from the lack of Azure.* dependencies in the TypeSpec instead of it being a flag

File: packages/rlc-common/src/buildLogger.ts
Comment (lirenhe) [inline packages/rlc-common/src/buildLogger.ts:13]: Could you explain why we need to disable logger for non-branded packages?

Comment (MaryGao) [reply packages/rlc-common/src/buildLogger.ts:13]: Yes, logger is removed in ts http runtime.

File: packages/typespec-ts/src/index.ts
Comment (joheredi) [reply packages/typespec-ts/src/index.ts:225]: I wouldn't block this PR on this, since probably we need to come up with an agreement cross language about how to detect this. but lets file an issue

Comment (MaryGao) [reply packages/typespec-ts/src/index.ts:225]: I filed an issue https://github.com/Azure/autorest.typescript/issues/2101, but my understanding is that given typespec we could generate both Azure SDK and Non-Azure SDK as output. If we calculate them automatically how to implement that?

File: packages/rlc-common/src/buildClient.ts
Comment (MaryGao) [inline packages/rlc-common/src/buildClient.ts:334]: Offline dicussed with @joheredi We may prefer to remove the option `addCredentials` from TypeSpec generation because TypeSpec could express no credential defined.

I will have a separated pr for this.

File: packages/typespec-test/test/openai/spec/routes.tsp
Comment (MaryGao) [inline packages/typespec-test/test/openai/spec/routes.tsp:4]: Update the OpenAI RLC with the latest spec.

File: packages/autorest.typescript/src/restLevelClient/transforms/transform.ts
Comment (lirenhe) [inline packages/autorest.typescript/src/restLevelClient/transforms/transform.ts:17]: build3rdPartyImports or buildNonbrandingImports

File: packages/rlc-common/src/buildResponseTypes.ts
Comment (lirenhe) [inline packages/rlc-common/src/buildResponseTypes.ts:88]: I saw similar code repeating multiple time, can we refactor this part?

File: packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/src/UsageClient.ts
Comment (lirenhe) [inline packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/src/UsageClient.ts:1]: Do we have a way to replace this?

File: packages/autorest.typescript/src/restLevelClient/transforms/transform.ts
Comment (MaryGao) [reply packages/autorest.typescript/src/restLevelClient/transforms/transform.ts:17]: The `third party` is mainly for SDK's dependency not for branding or non-branding. I sperated the inner imports which means the imports from the SDK iteself e.g models or parameters ect and third party imports which means the depedency outside of the SDK.

So both the `@azure/core-rest-pipeline` or `@typespec/ts-http-runtime` are third parties.

File: packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/tspconfig.yaml
Comment (lirenhe) [inline packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/tspconfig.yaml:1]: why do we need this file?

File: packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/src/UsageClient.ts
Comment (MaryGao) [reply packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/src/UsageClient.ts:1]: Currently no, we hard coded this. But we could replace them, any suggestions or default copyright info for non branded?

File: packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/tspconfig.yaml
Comment (MaryGao) [reply packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/tspconfig.yaml:1]: This file is not generated, this is the config file for generation and the input of `tsp compile`. We enable this in integration so that we could have case-level controls for different options which could help us verify the combination of various setting in integration.

File: packages/rlc-common/src/buildResponseTypes.ts
Comment (MaryGao) [reply packages/rlc-common/src/buildResponseTypes.ts:88]: Let me build a helper for this.

File: packages/typespec-test/test/openai_non_branded/generated/typespec-ts/package.json
Comment (lirenhe) [inline packages/typespec-test/test/openai_non_branded/generated/typespec-ts/package.json:4]: Can we also make this configurable?

Comment (lirenhe) [reply packages/typespec-test/test/openai_non_branded/generated/typespec-ts/package.json:4]: It is better no to have the keyword of Microsoft either.

File: packages/rlc-common/src/buildResponseTypes.ts
Comment (MaryGao) [reply packages/rlc-common/src/buildResponseTypes.ts]: updated

File: packages/typespec-test/test/openai_non_branded/generated/typespec-ts/package.json
Comment (MaryGao) [reply packages/typespec-test/test/openai_non_branded/generated/typespec-ts/package.json:4]: updated

File: packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/src/UsageClient.ts
Comment (MaryGao) [reply packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/src/UsageClient.ts]: I removed the line for `// Copyright (c) Microsoft Corporation.` in non-branded.

File: packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/tspconfig.yaml
Comment (MaryGao) [reply packages/typespec-ts/test/nonBrandedIntegration/modular/generated/models/usage/tspconfig.yaml:1]: Remove the azureSDKforJS config

File: packages/typespec-ts/src/index.ts
Comment (lirenhe) [reply packages/typespec-ts/src/index.ts:225]: Currently, dotnet SDK is also using a flag to branding/unbranding.  So, I think it is fine for now to quickly unblock the demo, but we need to refactor it later.

File: packages/rlc-common/src/helpers/importsUtil.ts
Comment (joheredi) [inline packages/rlc-common/src/helpers/importsUtil.ts:8]: nit: I would prefer to call this `buildThirdPartyImports`

Comment (joheredi) [reply packages/rlc-common/src/helpers/importsUtil.ts:8]: Actually can we just call this buildCoreImports or buildRuntimeImports, I feel like 3rd party is a bit misleading. I would expect to see things external to Microsoft in here like lodash or axios when reading 3rd party

File: packages/typespec-ts/src/transform/transform.ts
Comment (joheredi) [inline packages/typespec-ts/src/transform/transform.ts:94]: If I understand this correctly, inner imports track the imports within the same library while thirdPartyImports tracks the imports of core/runtime libraries right?

nit: can we make this a single property imports: { internalImports, externalImports };

File: packages/typespec-ts/src/modular/buildOperations.ts
Comment (joheredi) [inline packages/typespec-ts/src/modular/buildOperations.ts:14]: It strikes me as odd to see an rlc-common import in the modular codegen layer. Probably okay but wanted to bring this up in case there is an alternative

File: packages/typespec-ts/src/modular/buildProjectFiles.ts
Comment (joheredi) [inline packages/typespec-ts/src/modular/buildProjectFiles.ts:494]: it would be great if we can encapsulate this logic and share it between the branded and unbranded codepaths.

**Review by joheredi — APPROVED** [review]
Looks great! Left a couple of comments, non blocking but would be great if we can address them before merging

File: packages/rlc-common/src/helpers/importsUtil.ts
Comment (MaryGao) [reply packages/rlc-common/src/helpers/importsUtil.ts:8]: Got it and update to buildRuntimeImports

File: packages/typespec-ts/src/modular/buildOperations.ts
Comment (MaryGao) [reply packages/typespec-ts/src/modular/buildOperations.ts:14]: Good catch up! Let me update this to modular.

File: packages/typespec-ts/src/transform/transform.ts
Comment (MaryGao) [reply packages/typespec-ts/src/transform/transform.ts:94]: update to  { internalImports, runtimeImports };

File: packages/typespec-ts/src/modular/buildProjectFiles.ts
Comment (MaryGao) [reply packages/typespec-ts/src/modular/buildProjectFiles.ts:494]: updated

**Pull request merged**
